presto-native-execution/presto_cpp/main/SystemConnector.cpp

```diff
@@ -0,0 +1,389 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/SystemConnector.h"
+#include "presto_cpp/main/PrestoTask.h"
+#include "presto_cpp/main/TaskManager.h"
+
+#include "velox/type/Timestamp.h"
+
+namespace facebook::presto {
+
+using namespace velox;
+
+namespace {
+
+static const std::string kRuntimeSchema = "runtime";
+static const std::string kTasksTable = "tasks";
+
+static std::vector<std::string> kTaskColumnNames = {
```

Comment:
Should we make the `kTaskColumnNames` and `kTaskColumnTypes` to be local variables in `SystemTableHandle::taskSchema` to avoid potential use-after-move?

Reply:
Good idea. Thanks !